core/src/main/java/com/yandex/yoctodb/util/mutable/impl/AscendingBitSetIndexToIndexMultiMap.java

```diff
@@ -0,0 +1,86 @@
+package com.yandex.yoctodb.util.mutable.impl;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import com.yandex.yoctodb.util.OutputStreamWritable;
+import com.yandex.yoctodb.util.mutable.ArrayBitSet;
+import com.yandex.yoctodb.util.mutable.IndexToIndexMultiMap;
+import com.yandex.yoctodb.v1.V1DatabaseFormat;
+import net.jcip.annotations.Immutable;
+import net.jcip.annotations.NotThreadSafe;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+
+/**
+ * {@link IndexToIndexMultiMap} implementation based on grouped {@link LongArrayBitSet}s
+ * For more information see {@link com.yandex.yoctodb.util.immutable.impl.AscendingBitSetIndexToIndexMultiMap}
+ *
+ * @author Andrey Korzinev (goodfella@yandex-team.ru)
+ */
+
+@Immutable
+@NotThreadSafe
+public class AscendingBitSetIndexToIndexMultiMap implements IndexToIndexMultiMap, OutputStreamWritable {
+    private final int documentsCount;
+    @NotNull
+    private final Collection<? extends Collection<Integer>> map;
+
+    public AscendingBitSetIndexToIndexMultiMap(
+            @NotNull final Collection<? extends Collection<Integer>> map,
+            final int documentsCount) {
+        if (documentsCount < 0)
+            throw new IllegalArgumentException("Negative document count");
+
+        this.map = map;
+        this.documentsCount = documentsCount;
+    }
+
+    @Override
+    public long getSizeInBytes() {
+        return 4L + // Type
+                4L + // Keys count
+                4L + // Bit set size in longs
+                8L * (map.size() + 1) * LongArrayBitSet.arraySize(documentsCount);
```

Comment:
Could you please add comment clarifying calculations in the end?

Reply:
Done!